File: Gemstones.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'gemstones' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts STRING_ARRAY arr as parameter.
     */

    public static int gemstones(List<string> arr)
    {
        var common = new BitArray(26, true);

        foreach (var rock in arr)
        {
            var current = new BitArray(26);
            foreach (char c in rock)
            {
                current[c - 'a'] = true;
            }
            common.And(current);
        }

        int count = 0;
        foreach (bool b in common)
            if (b) count++;

        return count;
    }
    
    
    public static int PopCount(int n)
    {
        int count = 0;
        while (n != 0)
        {
            n &= (n - 1);
            count++;
        }
        return count;
    }

    public static int gemstones2(List<string> arr)
    {
        int common = (1 << 26) - 1;

        foreach (var rock in arr)
        {
            int mask = 0;
            foreach (char c in rock)
            {
                mask |= 1 << (c - 'a');
            }
            common &= mask;
        }

        return PopCount(common);
    }
    
    
    public static int gemstones3(List<string> arr)
    {
        int gemsCount = 0;
        var mp = new Dictionary<int, int>();

        foreach (var rock in arr)
        {
            var st = new HashSet<int>();
            foreach (char c in rock)
            {
                st.Add(c - 'a');
            }

            foreach (int ch in st)
            {
                if (mp.ContainsKey(ch))
                    mp[ch]++;
                else
                    mp[ch] = 1;
            }
        }

        foreach (var pair in mp)
        {
            if (pair.Value == arr.Count)
                gemsCount++;
        }

        return gemsCount;
    }
}


class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int n = Convert.ToInt32(Console.ReadLine().Trim());

        List<string> arr = new List<string>();

        for (int i = 0; i < n; i++)
        {
            string arrItem = Console.ReadLine();
            arr.Add(arrItem);
        }

        int result = Result.gemstones(arr);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Gemstones.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'gemstones' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts STRING_ARRAY arr as parameter.
     */

    public static int gemstones(List<String> arr) {
        BitSet common = new BitSet(26);
        common.set(0, 26);

        for (String rock : arr) {
            BitSet current = new BitSet(26);
            for (char c : rock.toCharArray()) {
                current.set(c - 'a');
            }
            common.and(current);
        }

        return common.cardinality();
    }


    public static int popCount(int n) {
        int count = 0;
        while (n != 0) {
            n &= (n - 1);
            count++;
        }
        return count;
    }

    public static int gemstones2(List<String> arr) {
        int common = (1 << 26) - 1;

        for (String rock : arr) {
            int mask = 0;
            for (char c : rock.toCharArray()) {
                mask |= 1 << (c - 'a');
            }
            common &= mask;
        }

        return popCount(common);
    }


    public static int gemstones3(List<String> arr) {
        int gemsCount = 0;
        Map<Integer, Integer> mp = new HashMap<>();

        for (String rock : arr) {
            Set<Integer> st = new HashSet<>();
            for (char c : rock.toCharArray()) {
                st.add(c - 'a');
            }

            for (int ch : st) {
                mp.put(ch, mp.getOrDefault(ch, 0) + 1);
            }
        }

        for (int count : mp.values()) {
            if (count == arr.size()) {
                gemsCount++;
            }
        }

        return gemsCount;
    }
}


public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<String> arr = IntStream.range(0, n).mapToObj(i -> {
            try {
                return bufferedReader.readLine();
            } catch (IOException ex) {
                throw new RuntimeException(ex);
            }
        })
            .collect(toList());

        int result = Result.gemstones(arr);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}
